# Correlation

Correlation is a statistical measure that tells how two variables or features are related to each other. In simple words, correlation explains
how would one variable behave when the value of other variable is increased or decreased.

The relation between the two variables can be 

1. __Positive__ - meaning an increase of one would increase the value of the other variable as well.
2. __Negative__ - the two variables are inversely related, an increase of one would decrease the other and vice versa
3. __No Relation__ - no affects of the value change on other

# Correlation coefficient

This statistic measure gives the idea of how strong the two features are correlated. This value ranges between -1 & 1.

-1 - indicates a strong negative relation

0 - indicates no relation

1 - indicates the strong relation

In the following  

(a) indicates strong positive correlation between x1 and x2

(b)indicates a stong negative correlation and where as

(c) indicates no correlation

<img src='corrs.png'/>

#  Pearson Correlation

Pearson correlation measures the linear relationship between two variables. 

if the Pearson correlation has value of 0 it means the two variables are not linearly related but there can be other non-linear relationship between them. Hence, it should not be concluded that two variables are not related just by considering the pearson value of 0.



Pearson Correlation between variables x & y =    __covariance(x,y) / (stdDev(x)*stdDev(y))__

# Calculating Covariance using numpy

In [20]:
import numpy as np

x = [10,13,15,17,20]
y = [76,80,84,86,90]

cov_matrix = np.cov(x,y)

print(cov_matrix)

[[14.5 20.5]
 [20.5 29.2]]


 The above displays the covariance matrix in the following format

  cov(x,x)    cov(x,y)

  cov(y,x)    cov(y,y)
  
  we are interested to find the covariance of x & y cov(x,y)
  
  __Note__ cov(x,y) = cov(y,x)

In [21]:
 print('covariance of x & y = ',cov_matrix[0,1])

covariance of x & y =  20.5


#  calculating Covariance by hand

Let us consider the same values used in the above step

x = [10,13,15,17,20]

y = [76,80,84,86,90]

<img src='cov_2.jpeg' width='500' height='500'/>
<img src='cov_1.jpeg' width='500' height='500'/>


# calculating pearson Correlation

In [22]:
print("Pearson Correlation is denoted  by r")
print("Pearson Correlation between x & y = r  =", np.corrcoef(x,y)[0,1])


Pearson Correlation is denoted  by r
Pearson Correlation between x & y = r  = 0.9962731687323246


# calculating pearson Correlation by hand

we have already calculated the cov(x,y) = 20.5

pearson correlation = cov(x,y) / stdDev(x)* stdDev(y)

<img src='pearson.jpeg'  width='500' height='200'/>